src/Riverline/MultiPartParser/Part.php

```diff
@@ -46,7 +46,7 @@ public function __construct($content)
     protected function parseContentString($content)
     {
         // Split headers and body
-        $splits = preg_split('/(\r?\n){2}/', $content, 2);
```

Comment:
Note that I recently had an issue due to the fact that the \r is optional.  The following piece of code is used to split a body in its parts: `preg_match('/'.$separator.'\r?\n(.+)\r?\n'.$separator.'--/s', $body, $matches)`.  The greedy `(.+)` matcher will eat up the optional `\r` following it. As a result the body of the part contains an extra `\r`. This causes problems when a digest or signature for verification over that body is calculated later on.
An easy fix for this could be to make the `+` non-greedy an use `(.+?)`.